src/pages/login/signup.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import './login.css';
+import { Link, Navigate, useNavigate } from 'react-router-dom';
+import { useAuth } from '../../context/authContext';
+
+export default function Signup() {
+    const navigate = useNavigate();
+    const [pswdType, setPswdType] = useState("password");
+    const [username, setUsername] = useState();
+    const [email, setEmail] = useState();
+    const [password, setPassword] = useState();
+    const [firstName, setFirstName] = useState();
+    const [lastName, setLastName] = useState();
```

Comment:
Suggestion : Here you have more than 3 states related to single entity . It would be better if we could maintain an object of these state. It becomes easier to manage.